.github/workflows/CICD.yml

```diff
@@ -0,0 +1,298 @@
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+    env:
+      extensions: bcmath, curl, dom, gd, imagick, json, libxml, mbstring, pcntl, pdo, pdo_sqlite, pdo_mysql, pdo_pgsql, pgsql, sqlite3, zip
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set Up Imagick, FFmpeg & Exiftools
+        run: |
+          sudo apt-get update
+          sudo apt-get --fix-broken install
+          sudo apt-get -y install ffmpeg libimage-exiftool-perl
+
+      - name: setup Databases
+        run: |
+          sudo service mysql start
+          touch database/database.sqlite
+          mysql -uroot -proot -e 'create database homestead_test;'
+
+      - name: Setup PHP Action
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: ${{ env.extensions }}
+          tools: pecl, composer
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: --no-dev
+
+      - name: Build Dist
+        run: |
+          make dist-clean
+
+      - name: copy Env
+        run: |
+          cp .github/workflows/.env.${{ matrix.sql-versions }} Lychee/.env
+
+      - name: setup MySQL Database with v3
+        run: |
+          mysql -uroot -proot homestead_test < .github/workflows/v3.sql
+
+      - name: Generate secure key & Migrate & go backward
+        run: |
+          cd Lychee
+          pwd
+          php artisan key:generate
+          php artisan migrate
+          php artisan migrate:rollback
+
+  createArtifact:
+    name: 3️⃣ Build Artifact
+    needs:
+      - phpstan
+      - dist
+      - tests
+    runs-on: ubuntu-latest
+    if: (github.event_name == 'push' && github.event.pull_request.head.repo.full_name != github.repository && startsWith(github.ref, 'refs/heads/master'))
+    env:
+      extensions: bcmath, curl, dom, gd, imagick, json, libxml, mbstring, pcntl, pdo, pdo_sqlite, pdo_mysql, pdo_pgsql, pgsql, sqlite3, zip
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set Up Imagick, FFmpeg & Exiftools
+        run: |
+          sudo apt-get update
+          sudo apt-get --fix-broken install
+          sudo apt-get -y install ffmpeg libimage-exiftool-perl
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
+          extensions: ${{ env.extensions }}
+          coverage: none
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: --no-dev
+
+      - name: Build Dist
+        run: |
+          make clean dist
+
+      - name: Upload a Build Artifact
+        uses: actions/upload-artifact@v3.1.0
+        with:
+          name: Lychee-Dist.zip
+          path: Lychee.zip
```

Comment:
Where is the resulting ZIP archive finally put? Inside the branch, if yes, where or in a different location?

Reply:
Found it myself: https://github.com/actions/upload-artifact#where-does-the-upload-go